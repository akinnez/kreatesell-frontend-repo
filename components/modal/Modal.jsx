import { useState, useEffect } from "react";
import Image from "next/image";
import { CloseIcon } from "../../utils";
import styles from "./Modal.module.scss";

export const Modal = ({
	children,
	visible,
	onClose,
	containerStyle,
	cancelPropagation = false,
	closeButton = false,
	closeBtnAction,
	...rest
}) => {
	const [_visible, setVisible] = useState(false);

	useEffect(() => {
		setVisible(visible);
	}, [visible]);

	return (
		<div
			{...rest}
			className={`
			${styles.modal}
			${!_visible && "hidden"} 
			${_visible && styles.styleDisplay} 
			${rest.className}
			`}
			onClick={() => (cancelPropagation ? null : onClose && onClose())}
		>
			<div
				className={`${styles.container} ${containerStyle}`}
				onClick={(e) => e.stopPropagation()}
			>
				{closeButton && (
					<div
						className={styles.closeIcon}
						onClick={closeBtnAction || setVisible((value) => !value)}
					>
						<Image src={CloseIcon} />
					</div>
				)}
				{visible && <div className={styles.modalContent}>{children}</div>}
			</div>
		</div>
	);
};
