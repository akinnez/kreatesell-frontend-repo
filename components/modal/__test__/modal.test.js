import renderer from "react-test-renderer";
import { Modal } from "../Modal";

const handleClose = jest.fn();
describe("Modal : ", () => {
  it("renders a snapshot of <Modal /> with mock props", () => {
    const modal = renderer.create(<Modal />).toJSON();
    expect(modal).toMatchSnapshot();
  });
  it("renders a snapshot of <Modal /> with mock props and closeButton and cancelPropagation both at true and visible set at false", () => {
    const modal = renderer
      .create(
        <Modal
          onClose={handleClose}
          closeButton={false}
          cancelPropagation={false}
          visible={false}
        />
      )
      .toJSON();
    expect(modal).toMatchSnapshot();
  });
  it("renders a snapshot of <Modal /> with mock props and closeButton set at false and cancelPropagation at false,  visible set at false", () => {
    const modal = renderer
      .create(
        <Modal
          onClose={handleClose}
          closeButton={false}
          cancelPropagation={true}
          visible={false}
        />
      )
      .toJSON();
    expect(modal).toMatchSnapshot();
  });
  it("renders a snapshot of <Modal /> with mock props and closeButton set at false and cancelPropagation at false,  visible set at true", () => {
    const modal = renderer
      .create(
        <Modal
          onClose={handleClose}
          closeButton={false}
          cancelPropagation={false}
          visible={true}
        />
      )
      .toJSON();
    expect(modal).toMatchSnapshot();
  });
  it("renders a snapshot of <Modal /> with mock props and closeButton set at false and cancelPropagation at true,  visible set at true", () => {
    const modal = renderer
      .create(
        <Modal
          onClose={handleClose}
          closeButton={false}
          cancelPropagation={true}
          visible={true}
        />
      )
      .toJSON();
    expect(modal).toMatchSnapshot();
  });

  // @btnClose = true, too many re-renders happen.
});
