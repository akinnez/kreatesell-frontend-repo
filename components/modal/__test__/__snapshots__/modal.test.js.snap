// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal :  renders a snapshot of <Modal /> with mock props 1`] = `
<div
  className="
			modal
			hidden 
			false 
			undefined
			"
  onClick={[Function]}
>
  <div
    className="container undefined"
    onClick={[Function]}
  />
</div>
`;

exports[`Modal :  renders a snapshot of <Modal /> with mock props and closeButton and cancelPropagation both at true and visible set at false 1`] = `
<div
  className="
			modal
			hidden 
			false 
			undefined
			"
  onClick={[Function]}
>
  <div
    className="container undefined"
    onClick={[Function]}
  />
</div>
`;

exports[`Modal :  renders a snapshot of <Modal /> with mock props and closeButton set at false and cancelPropagation at false,  visible set at false 1`] = `
<div
  className="
			modal
			hidden 
			false 
			undefined
			"
  onClick={[Function]}
>
  <div
    className="container undefined"
    onClick={[Function]}
  />
</div>
`;

exports[`Modal :  renders a snapshot of <Modal /> with mock props and closeButton set at false and cancelPropagation at false,  visible set at true 1`] = `
<div
  className="
			modal
			hidden 
			false 
			undefined
			"
  onClick={[Function]}
>
  <div
    className="container undefined"
    onClick={[Function]}
  >
    <div
      className="modalContent"
    />
  </div>
</div>
`;

exports[`Modal :  renders a snapshot of <Modal /> with mock props and closeButton set at false and cancelPropagation at true,  visible set at true 1`] = `
<div
  className="
			modal
			false 
			styleDisplay 
			undefined
			"
  onClick={[Function]}
>
  <div
    className="container undefined"
    onClick={[Function]}
  >
    <div
      className="modalContent"
    />
  </div>
</div>
`;
